sdks/java/extensions/protobuf/src/main/java/org/apache/beam/sdk/extensions/protobuf/ProtoDynamicMessageSchema.java

```diff
@@ -0,0 +1,854 @@
+        case "google.protobuf.BoolValue":
+        case "google.protobuf.Int64Value":
+        case "google.protobuf.Int32Value":
+        case "google.protobuf.UInt64Value":
+        case "google.protobuf.UInt32Value":
+          return new WrapperConvert(field, new PrimitiveConvert(valueField));
+        case "google.protobuf.BytesValue":
+          return new WrapperConvert(field, new BytesConvert(valueField));
+        case "google.protobuf.Timestamp":
+        case "google.protobuf.Duration":
+          // handled by logical type case
+          break;
+      }
+    }
+    switch (fieldType.getTypeName()) {
+      case BYTE:
+      case INT16:
+      case INT32:
+      case INT64:
+      case FLOAT:
+      case DOUBLE:
+      case STRING:
+      case BOOLEAN:
+        return new PrimitiveConvert(field);
+      case BYTES:
+        return new BytesConvert(field);
+      case ARRAY:
+      case ITERABLE:
+        return new ArrayConvert(this, field);
+      case MAP:
+        return new MapConvert(this, field);
+      case LOGICAL_TYPE:
+        String identifier = field.getType().getLogicalType().getIdentifier();
+        switch (identifier) {
+          case ProtoSchemaLogicalTypes.Fixed32.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.Fixed64.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.SFixed32.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.SFixed64.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.SInt32.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.SInt64.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.UInt32.IDENTIFIER:
+          case ProtoSchemaLogicalTypes.UInt64.IDENTIFIER:
+            return new LogicalTypeConvert(field, fieldType.getLogicalType());
+          case NanosInstant.IDENTIFIER:
+            return new TimestampConvert(field);
+          case NanosDuration.IDENTIFIER:
+            return new DurationConvert(field);
+          case EnumerationType.IDENTIFIER:
+            return new EnumConvert(field, fieldType.getLogicalType());
+          case OneOfType.IDENTIFIER:
+            return new OneOfConvert(this, field, fieldType.getLogicalType());
+          default:
+            throw new IllegalStateException("Unexpected logical type : " + identifier);
+        }
+      case ROW:
+        return new MessageConvert(this, field);
+      default:
+        throw new IllegalStateException("Unexpected value: " + fieldType);
+    }
+  }
+
+  private List<Convert> createConverters(Schema schema) {
+    List<Convert> fieldOverlays = new ArrayList<>();
+    for (Schema.Field field : schema.getFields()) {
+      fieldOverlays.add(createConverter(field));
+    }
+    return fieldOverlays;
+  }
+
+  public Schema getSchema() {
+    return context.getSchema();
+  }
+
+  public SchemaCoder<T> getSchemaCoder() {
+    return SchemaCoder.of(
+        context.getSchema(),
+        TypeDescriptor.of(context.getBaseClass()),
+        toRowFunction,
+        fromRowFunction);
+  }
```

Comment:
I don't see this function being used anywhere. Remove?

Reply:
I used it in a production pipeline, but as it's not available in the other providers I removed it, (squashed and rebased).